temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowRunTaskHandler.java

```diff
@@ -255,10 +255,14 @@ private void applyServerHistory(long lastEventId, WorkflowHistoryIterator histor
               implementationOptions.getFailWorkflowExceptionTypes();
           for (Class<? extends Throwable> failType : failTypes) {
             if (failType.isAssignableFrom(e.getClass())) {
+              metricsScope.counter(MetricsType.WORKFLOW_FAILED_COUNTER).inc(1);
```

Comment:
I see `completeWorkflow` increments this metric. Do these exceptions not make it to `completeWorkflow`? Is there a common place that `completeWorkflow`-based exceptions _and_ these are turned to workflow failures that may be a better place to move the metric, e.g. `WorkflowStateMachines.failWorkflow`? Or even if we don't want to change `completeWorkflow`, is there a common place where `WorkflowExecutionException` is caught and converted?

Reply:
>Do these exceptions not make it to completeWorkflow

Correct, this code catches exceptions from the state machine, not the workflow code itself

>Is there a common place that completeWorkflow-based exceptions and these are turned to workflow failures that may be a better place to move the metric, e.g. WorkflowStateMachines.failWorkflow

No

>is there a common place where WorkflowExecutionException is caught and converted?

The only place would be when the SDK responds to the server, but that would require changing where all completion metrics are emitted. That work is captured here https://github.com/temporalio/sdk-java/issues/1590. I was hopping to defer that work till later.